False positive "[TypedDict instance] is not callable"

### Steps to reproduce

Given a file `a.py`:
```python
from typing import TypedDict

Link = TypedDict("Link", {"href": str})
Link(href="foo")
```

### Current behavior

Result of `pylint a.py`:
```
************* Module a
a.py:4:0: E1102: Link is not callable (not-callable)

------------------------------------
Your code has been rated at -6.67/10
```

### Expected behavior

This should not be an error. `TypedDict` is a subclass of `dict`, so line 4 is equivalent to `dict(href="foo")`. The code also runs fine and passes `mypy --strict a.py`.

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0-a4
astroid 2.6.2
Python 3.8.10 (default, Jul  7 2021, 14:29:06) 
[GCC 11.1.0]
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -56,6 +56,10 @@ Release date: TBA
 
   Closes #3525
 
+* Fix false-positive ``not-callable`` with alternative ``TypedDict`` syntax
+
+  Closes #4715
+
 * Clarify documentation for consider-using-from-import
 
 * Don't emit ``unreachable`` warning for empty generator functions

--- a/tests/functional/t/typedDict.py
+++ b/tests/functional/t/typedDict.py
@@ -23,3 +23,9 @@ CustomTD5 = TypedDict("CustomTD5", {"var5": bool})
 my_dict = CustomTD(var=1)
 my_dict["var"]
 my_dict["var"] = 2
+
+
+# https://github.com/PyCQA/pylint/issues/4715
+# Instance of TypedDict should be callable
+Link = TypedDict("Link", {"href": str})
+Link(href="foo")

